src/UserSpaceInstrumentation/InstrumentProcess.cpp

```diff
@@ -103,6 +109,124 @@ bool IsBlocklisted(std::string_view function_name) {
+// halted.
+ErrorMessageOr<MachineCode> MachineCodeForCloneCall(pid_t pid, void* library_handle,
+                                                    uint64_t top_of_stack) {
+  constexpr uint64_t kCloneFlags =
+      CLONE_FILES | CLONE_FS | CLONE_IO | CLONE_SIGHAND | CLONE_SYSVSEM | CLONE_THREAD | CLONE_VM;
+  constexpr uint32_t kSyscallNumberClone = 0x38;
+  constexpr uint32_t kSyscallNumberExit = 0x3c;
+  constexpr const char* kInitializeInstrumentationFunctionName = "InitializeInstrumentation";
+  OUTCOME_TRY(void* initialize_instrumentation_function_address,
+              DlsymInTracee(pid, library_handle, kInitializeInstrumentationFunctionName));
+  MachineCode code;
+  code.AppendBytes({0x48, 0xbf})
+      .AppendImmediate64(kCloneFlags)  // mov kCloneFlags, rdi
+      .AppendBytes({0x48, 0xbe})
+      .AppendImmediate64(top_of_stack)  // mov top_of_stack, rsi
+      .AppendBytes({0x48, 0xba})
+      .AppendImmediate64(0x0)  // mov parent_tid, rdx
+      .AppendBytes({0x49, 0xba})
+      .AppendImmediate64(0x0)  // mov child_tid, r10
+      .AppendBytes({0x49, 0xb8})
+      .AppendImmediate64(0x0)           // mov tls, r8
+      .AppendBytes({0x48, 0xc7, 0xc0})  // mov kSyscallNumberClone, rax
+      .AppendImmediate32(kSyscallNumberClone)
+      .AppendBytes({0x0f, 0x05})                          // syscall (clone)
+      .AppendBytes({0x48, 0x85, 0xc0})                    // testq	rax, rax
+      .AppendBytes({0x0f, 0x84, 0x01, 0x00, 0x00, 0x00})  // jz 0x01(rip)
+      .AppendBytes({0xcc})                                // int3
+      .AppendBytes({0x48, 0xb8})
+      .AppendImmediate64(absl::bit_cast<uint64_t>(
+          initialize_instrumentation_function_address))  // mov initialize_instrumentation, rax
+      .AppendBytes({0xff, 0xd0})                         // call rax
+      .AppendBytes({0x48, 0xc7, 0xc7, 0x00, 0x00, 0x00, 0x00})  // mov 0x0, rdi
+      .AppendBytes({0x48, 0xc7, 0xc0})                          // mov kSyscallNumberExit, rax
+      .AppendImmediate32(kSyscallNumberExit)
+      .AppendBytes({0x0f, 0x05});  // syscall (exit)
+  return code;
+}
+
+ErrorMessageOr<void> WaitForThreadToExit(pid_t pid, pid_t tid) {
+  // In all tests the thread exited in one to three rounds of waiting for one millisecond. To make
+  // sure that we never stall OrbitService here we return an error when the thread requires an
+  // excessive amount of time to exit.
+  constexpr int kNumberOfRetries = 3000;
+  int count = 0;
+  auto tids = orbit_base::GetTidsOfProcess(pid);
+  while (std::find(tids.begin(), tids.end(), tid) != tids.end()) {
+    if (count++ > kNumberOfRetries) {
+      return ErrorMessage("Initilization thread injected into target process failed to exit.");
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds{1});
+    tids = orbit_base::GetTidsOfProcess(pid);
+  }
+  return outcome::success();
+}
+
+// These are the names of the threads that we will be spawned when
```

Comment:
we will be spawned -> "we will spawn" / "will be spawned"

Reply:
ahh, copy pasted the error....